crates/client/rpc-core/src/utils.rs

```diff
@@ -87,17 +89,19 @@ pub fn to_deploy_account_tx(tx: BroadcastedDeployAccountTransaction) -> Result<D
         .try_into()
         .map_err(|_| anyhow!("failed to bound signatures Vec<H256> by MaxArraySize"))?;
 
-    let sender_address = calculate_contract_address(
-        ContractAddressSalt(StarkFelt(contract_address_salt)),
-        ClassHash(StarkFelt(account_class_hash)),
-        &Calldata(calldata.into()),
-        StarknetContractAddress::default(),
+    let sender_address = Felt252Wrapper::try_from(
+        &calculate_contract_address(
+            ContractAddressSalt(StarkFelt(contract_address_salt)),
+            ClassHash(StarkFelt(account_class_hash)),
+            &Calldata(calldata.into()),
+            StarknetContractAddress::default(),
+        )
+        .map_err(|e| anyhow!("Failed to calculate contract address: {e}"))?
+        .0
+        .0
+        .0,
```

Comment:
current types are 
<img width="104" alt="image" src="https://github.com/keep-starknet-strange/madara/assets/18620296/f0dfcd19-61ce-494a-a09b-a062e4a54983">
so we could stop at `StarkFelt` and `.into`